File: Reverse a Linked List in groups of given size[21-July-2023]/Iterative/Solution.py
#  >> PYTHON ITERATIVE APPROACH

class Solution:
    def reverse(self,head, k):
        def reverseList(head):
            prev=None
            next=None
            ptr=head
            
            while(ptr):
                next=ptr.next
                ptr.next=prev
                prev=ptr
                ptr=next
        
            return prev
            
        nn=head
        ans=Node(-1)
        ansPtr=ans
        
        while(nn!=None):
            start=nn
            end=nn
            
            cnt=k
            while(end!=None and cnt!=1):
                end=end.next
                cnt-=1
            
            if(end!=None):
                nn=end.next
                end.next=None
            else:
                nn=None
            ansPtr.next=reverseList(start)
            ansPtr=start
        
        return ans.next

File: Reverse a Linked List in groups of given size[21-July-2023]/Iterative/Solution.cpp
//  >> C++ ITERATIVE APPROACH <<

class Solution
{
    public:
    struct node* reverseList(struct node *head) {
        struct node* prev = nullptr;
        struct node* next = nullptr;
        struct node* ptr = head;

        while (ptr) {
            next = ptr->next;
            ptr->next = prev;
            prev = ptr;
            ptr = next;
        }

        return prev;
    }

    struct node *reverse (struct node *head, int k)
    { 
        struct node* nn = head;
        struct node* ans =new struct node(-1);
        struct node* ansPtr = ans;

        while (nn != nullptr) {
            struct node* start = nn;
            struct node* end = nn;

            int cnt = k;
            while (end != nullptr && cnt != 1) {
                end = end->next;
                cnt--;
            }

            if (end != nullptr) {
                nn = end->next;
                end->next = nullptr;
            } else {
                nn = nullptr;
            }
            
            ansPtr->next = reverseList(start);
            ansPtr = start;
        }

        return ans->next;
    }
};
